src/devices/Pca95x4/Pca95x4.cs

```diff
@@ -0,0 +1,199 @@
+
+using System;
+using System.Device.Gpio;
+using System.Device.I2c;
+
+namespace Iot.Device.Pca95x4
+{
+    /// <summary>
+    /// A general purpose parallel I/O expansion for I2C applications.
+    /// </summary>
+    public class Pca95x4 : IDisposable
+    {
+        private readonly I2cDevice _i2cDevice;
+        private GpioController _masterGpioController;
+        private readonly int? _interrupt;
+
+        /// <summary>
+        /// Initializes new instance of Pca95x4.
+        /// A general purpose parallel I/O expansion for I2C applications.
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device used for communication.</param>
+        /// <param name="interrupt">The input pin number that is connected to the interrupt (INT).</param>
+        public Pca95x4(I2cDevice i2cDevice, int? interrupt = null)
+        {
+            _i2cDevice = i2cDevice;
+            _interrupt = interrupt;
+
+            InitializeMasterGpioController();
+        }
+
+        private static void ValidateBitNumber(int bitNumber)
+        {
+            if (bitNumber < 0 || bitNumber > 7)
+            {
+                throw new IndexOutOfRangeException($"Invalid bit number {bitNumber}.");
+            }
+        }
+
+        private static void ClearBit(ref byte data, int bitNumber)
+        {
+            ValidateBitNumber(bitNumber);
+            data &= (byte)~(1 << bitNumber);
+        }
+
+        private void SetBit(ref byte data, int bitNumber)
+        {
+            ValidateBitNumber(bitNumber);
+            data |= (byte)(1 << bitNumber);
+        }
+
+        private static bool GetBit(byte data, int bitNumber)
+        {
+            ValidateBitNumber(bitNumber);
+            return ((data >> bitNumber) & 1) == 1;
+        }
+
+        private void InitializeMasterGpioController()
+        {
+            // Only need master controller if there is external pin provided.
+            if (_interrupt != null)
+            {
+                _masterGpioController = new GpioController();
+                _masterGpioController.OpenPin((int)_interrupt, PinMode.Input);
+            }
+        }
+
+        /// <summary>
+        /// Reads a byte from a register.
+        /// </summary>
+        /// <param name="register">The register to read.</param>
+        /// <returns>The data read from the register.</returns>
+        public byte Read(Register register)
+        {
+            byte[] data = Read(register, 1);
+            return data[0];
+        }
+
+        /// <summary>
+        /// Reads a number of bytes from a register. 
+        /// All data returned is from the selected register as Pca95x4 does not auto-increment addressing.
+        /// </summary>
+        /// <param name="register">The register to read.</param>
+        /// <param name="byteCount"></param>
+        /// <returns>The data read from the register.</returns>
+        public byte[] Read(Register register, byte byteCount)
```

Comment:
Is there a scenario where you would want to read more than one byte? To me You would always want to read/write to one of the 4 registers at one time.

Reply:
I was questioning this, but wasn't sure if a user wanted to do a burst of read for the input register.  We could remove this and only allow the Read byte and let it be up to the app how many times to read.  Plus, the user might want an interval between the number of times read and this code wouldn't allow that.  I'll remove later today.  Thx